### PR TITLE
[PLAYER-5605] Switch to next video causes the playback not start from begining

### PR DESCRIPTION
**Pull request by oSulyma**
**Ticket goal:** Next asset should start from beginning or it's previous time
**How PR achieves the goal:** When VC hets notification `OOOoyalaPlayerStateChanged` player started playback from withInitialTime of appropriate selection options. But sometimes selection option can wrong defined (because of scrolling I assume), that why I added method for checking if embed code of selection option matches embed code of player's current item.

**Affected repo-s:** Sample-Apps only
**Unit tests:** Unit test not added, there is no Unit tests target

### CODE REVIEW

**Comment by oSulyma** [comment]
@cacel  Please, pay attention to this PR, because you know about this SampleApp much more than other. 
Thank you

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift [hunk 1]
```diff
@@ -183,31 +179,30 @@ class MultiplePlayerViewController: UIViewController {
       let cellHeight = cellRect.height
       let ratio = visibleHeight / cellHeight
       if ratio > tmpRatio {
-        indexPath = collectionView.indexPath(for: cell)!
+        indexPathForVisibleCell = collectionView.indexPath(for: cell)!
```
Comment (maxkupetskii) [inline PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: Please, do not force unwrap optionals

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift [hunk 2]
```diff
@@ -222,8 +217,23 @@ class MultiplePlayerViewController: UIViewController {
     }
   }
   
+  private func checkSelectionOptionAndReplaceIfNeed(selectionOption: PlayerSelectionOption) -> PlayerSelectionOption {
+
```
Comment (maxkupetskii) [inline PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: Please, remove extra line

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift [hunk 3]
```diff
@@ -222,8 +217,23 @@ class MultiplePlayerViewController: UIViewController {
...
   }
   
+  private func checkSelectionOptionAndReplaceIfNeed(selectionOption: PlayerSelectionOption) -> PlayerSelectionOption {
+    
+    guard let currentItem = sharedPlayer.player.currentItem else {
+      return selectionOption
+    }
+    
+    if currentItem.embedCode != selectionOption.embedCode {
+      //print("❌ ❌ ❌ playerSelection Option embedCode [\(selectionOption.embedCode)]. player.currentItem.embedCode: \(String(describing: sharedPlayer.player.currentItem.embedCode))")
```
Comment (maxkupetskii) [inline PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: Please, remove commented line

**Review by maxkupetskii — CHANGES_REQUESTED** [review]

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift (hunk 1 above)
Comment (oSulyma) [reply PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: It's not my code, but agree, it's dangerous approach. Fixed

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift (hunk 2 above)
Comment (oSulyma) [reply PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: Done

File: PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift (hunk 3 above)
Comment (oSulyma) [reply PlaybackLab/MultiplePlayersScrollingViewSampleApp/MultiplePlayersScrollingViewSampleApp/ViewControllers/MultiplePlayerViewController.swift]: Done

**Review by maxkupetskii — APPROVED** [review]

**Review by DmitriiMaslov — APPROVED** [review]

**Pull request merged**